File: 228-summary-ranges/summary-ranges.py
class Solution(object):
    def summaryRanges(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        result = []
        n = len(nums)
        i = 0

        while i < n:
            start = nums[i]

            while i+1 < n and nums[i+1] == nums[i]+1:
                i +=1
            
            end = nums[i]

            if start == end:
                result.append(str(start))
            else:
                result.append(str(start)+"->"+str(end))
            
            i+=1
        
        return result

File: 228-summary-ranges/summary-ranges.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> result = new ArrayList<>();

        int n = nums.length;
        int i = 0;

        while(i<n){
            int start = nums[i];

            while (i+1 < n && nums[i+1] == nums[i]+1) {
                i++;
            }
            int end = nums[i];

            if (start == end) {
                result.add(String.valueOf(start));
            } else {
                result.add(start + "->" + end);
            }
            i++;
        }
        return result;
    }
}